src/app/pages/common/entity/entity-form/entity-form-embedded.component.ts

```diff
@@ -426,4 +435,20 @@ export class EntityFormEmbeddedComponent implements OnInit, OnDestroy, AfterView
   setHiddenFieldSets(fs: string[]): void {
     this.hiddenFieldSets = fs;
   }
+
+  getRelationType(config: FieldConfig): RelationConfig | null {
+    switch (config.type) {
+      case 'list':
+        const listConfig: FormListConfig = config;
+        return listConfig;
+      case 'dict':
+        const dictConfig: FormDictConfig = config;
+        return dictConfig;
+      case 'selectionlist':
+        const selectionListConfig: FormSelectionListConfig = config;
+        return selectionListConfig;
+      default:
+        return null;
+    }
```

Comment:
Same here. Additional type information that you are providing inside a method doesn't actually leave it, and you get `RelationConfig` out.